### PR TITLE
[OTE-784] Compliance roundtable address filter

### PR DESCRIPTION
**Pull request by jerryfan01234**
### Changelist
Restrict comlink compliance screen from upserting when address is not a dydx wallet with deposit
Restrict compliance check during compliance update roundtable to dydx wallet with depoit + add logging

### Test Plan
[Describe how this PR was tested (if applicable)]

### Author/Reviewer Checklist
- [ ] If this PR has changes that result in a different app state given the same prior state and transaction list, manually add the `state-breaking` label.
- [ ] If the PR has breaking postgres changes to the indexer add the `indexer-postgres-breaking` label.
- [ ] If this PR isn't state-breaking but has changes that modify behavior in `PrepareProposal` or `ProcessProposal`, manually add the label `proposal-breaking`.
- [ ] If this PR is one of many that implement a specific feature, manually label them all `feature:[feature-name]`.
- [ ] If you wish to for mergify-bot to automatically create a PR to backport your change to a release branch, manually add the label `backport/[branch-name]`.
- [ ] Manually add any of the following labels: `refactor`, `chore`, `bug`.

## Summary by CodeRabbit

- **New Features**
	- Introduced a new risk score constant for enhanced error handling in compliance checks.
	- Added a new parameter to improve compliance data retrieval based on wallet addresses.
	- Enhanced filtering capability for identifying addresses that returned a 404 error.

- **Bug Fixes**
	- Improved handling of invalid addresses in compliance checks to prevent unnecessary data entries.

- **Tests**
	- Implemented new test cases to verify compliance data filtering and address handling logic.

- **Documentation**
	- Updated method signatures and interfaces to reflect new parameters and functionalities.

### CODE REVIEW

File: indexer/packages/postgres/src/stores/compliance-table.ts [hunk 1]
```diff
@@ -70,6 +73,14 @@ export async function findAll(
     baseQuery = baseQuery.where(ComplianceDataColumns.blocked, blocked);
   }
 
+  if (addressInWalletsTable === true) {
+    baseQuery = baseQuery.innerJoin(
```
Comment (Christopher-Li) [inline indexer/packages/postgres/src/stores/compliance-table.ts:77]: nice I've never seen this done before!

File: indexer/services/roundtable/src/tasks/update-compliance-data.ts [hunk 2]
```diff
@@ -318,10 +319,18 @@ async function getComplianceData(
         return result.value;
       },
     ));
+    const responses404:
```
Comment (Christopher-Li) [inline indexer/services/roundtable/src/tasks/update-compliance-data.ts:322]: nit: best not to include numbers in variable names, `notFoundResponses`

File: indexer/services/roundtable/src/tasks/update-compliance-data.ts [hunk 3]
```diff
@@ -318,10 +319,18 @@ async function getComplianceData(
...
+    PromiseFulfilledResult<ComplianceClientResponse>[] = successResponses.filter(
+      (result: PromiseSettledResult<ComplianceClientResponse>):
+      result is PromiseFulfilledResult<ComplianceClientResponse> => {
+        return result.status === 'fulfilled' && result.value.riskScore === NOT_IN_BLOCKCHAIN_RISK_SCORE.toString();
```
Comment (Christopher-Li) [inline indexer/services/roundtable/src/tasks/update-compliance-data.ts:326]: this logic is kinda messy, up to you, but would it be worthwhile to further populate `ComplianceClientResponse`? We already support return an `undefined` riskScore, might be cleaner to add a `status` field and have that encompass all of the states of `blocked`, `NOT_IN_BLOCKCHAIN`, and `NO_RULES_TRIGGERED`

Comment (jerryfan01234) [reply indexer/services/roundtable/src/tasks/update-compliance-data.ts:326]: Yes i was actually thinking of adding a new field to CompliacenClientResponse to track the states.

Comment (jerryfan01234) [reply indexer/services/roundtable/src/tasks/update-compliance-data.ts:326]: Looked further into it. Modifying ComplianceClientResponse has ripple effects into all other providers. We can leave the new field nullable. At minimum the mocks for elliptic provider need to be changed. Will just stick to conditioning on riskScore and add comment explaining.

**Review by Christopher-Li — APPROVED** [review]

**Comment by jerryfan01234** [comment]
@Mergifyio backport release/indexer/v6.x

**Pull request merged**